### DISCUSSION TITLE
Labeling image segmentation.

### DISCUSSION THREAD

**Discussion by idkwid2022** [pytorch-ignite/code-generator#217] [Q&A] [answered]
Hi,

I want to ask about the code, I don't know where to ask since I don't know if it's bug or just my lack of understanding.

I'm using image segmentation template and on data.py there are lines that I don't understand

https://github.com/pytorch-ignite/code-generator/blob/f1372194000fb461c9e3eb7f4f85f43ad8f61a6c/src/templates/template-vision-segmentation/data.py#L32-L54

I guessed it's the categories/classes names, but I can't find any reference to the list `target_names`, even on the parent classes `VOCSegmentation`, `_VOCBase`, `VisionDataset`, and `Dataset`.

And that brings the next question, how do I label the segmentation result? There are no link between the label and the image dataset right?

Using CVAT, I got labelmap.txt which contains label and mask color, so I was thinking it's the link, but then I can't find any code reading the file.

So, I'm confused. Could anyone please help me with this?

Thank you.

**Comment by vfdev-5** [comment, marked as answer]
Good question, concerning target_names this image segmentation template was ported from ref example in ignite where we also plot Confusion Matrix : https://github.com/pytorch/ignite/blob/master/examples/references/segmentation/pascal_voc2012/main.py#L156
So yes, it is unused in this template.
Target names is basically a map between integer mask indices and names.

**Reply by idkwid2022** [reply]
Thanks for clearing that up. I guess I have to read ignite's ref example. :cold_sweat:
